fuel-core-interfaces/src/db.rs

```diff
@@ -672,51 +676,72 @@ pub mod helpers {
 
         fn insert(
             &mut self,
-            _key: &Bytes32,
-            _value: &Transaction,
+            key: &Bytes32,
+            value: &Transaction,
         ) -> Result<Option<Transaction>, Self::Error> {
-            unreachable!()
+            let arc = Arc::new(value.clone());
+
+            if let Some(previous_tx) = self.data.lock().tx.insert(*key, arc) {
```

Comment:
FYI: this method will return the old value (the replaced value) that was overwritten. That was my understanding after reading this:
https://github.com/FuelLabs/fuel-storage/blob/62f163dc21c712e2fa8dd9a6a76cf94f82ffc0d0/src/lib.rs#L27